### PR TITLE
Fixed Typos and grammatical mistakes in CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by Soaibkhan38**
I have updated the CONTRIBUTING.md file, fixed all the grammatical typos I encountered in the file. So, now it will increase the readability for new contributors.

Please merge this

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -54,7 +53,7 @@ instead.
 
 ## Contributing to Kong documentation and the Kong Hub
 
-Improving the documentation is a most welcome form of contribution to the Kong
+Improving the documentation is the most welcome form of a contribution to the Kong
```
Comment (h3xar0n) [inline CONTRIBUTING.md]: The sentence is grammatically correct and the usage is fine, please revert this change.

"A most welcome" is a common phrase. https://sentence.yourdictionary.com/welcome

"Form of [noun]" refers to a mode of an abstract concept, as opposed to "form of a [noun]," a specific event or entity.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -71,9 +70,8 @@ instead.
 When contributing, be aware of a few things:
 
 - Documentation for Kong Hub listings, which includes all Kong Inc.-published
-  and community-published plugins and integrations, lives in the `app/_hub`
-  and `app/_data/extensions` directories. **Versioning is optional, and thus
-  potentially inconsistent, for this part of the documentation**.
+  and community-published plugins and integrations lives in the `app/_hub`
+  and `app/_data/extensions` directories. **Versioning is optional, and thus potentially inconsistent, for this part of the documentation**.
```
Comment (h3xar0n) [inline CONTRIBUTING.md]: What is the purpose of this change?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -136,16 +134,15 @@ Adding a new listing to the Kong Hub may be proposed by:
 
 1. Clone this repo
 1. Create a publisher directory at`_app/_hub/`, such as
- `_app/_hub/your-github-handle` (if you are contributing as an individual)
- or `_app/_hub/company-name` (if you are contributing as a company). See
- other Kong Hub listings for examples of publisher names.
+ `_app/_hub/your-GitHub-handle` (if you are contributing as an individual)
+ or `_app/_hub/company-name` (if you are contributing as a company). See other Kong Hub listings for examples of publisher names.
```
Comment (h3xar0n) [inline CONTRIBUTING.md]: What is the purpose of this change?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -212,7 +209,7 @@ the existing plugins for examples, and see additional advice in
 In this repository, `master` represents the current state of the website, and
 should constantly be deployed. Branch off of this branch for local development.
 
-If you have write access to the GitHub repository, please follow the following
+If you have to write access to the GitHub repository, please follow the following
```
Comment (h3xar0n) [inline CONTRIBUTING.md]: "If you have write access" is correct, not "to write."

Write access is a concept in computing that refers to the permission of a user to "write" data to memory. Compare it with "read access," etc.

Grammarly and other spellcheckers do not understand this, but human readers of documentation should.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -227,24 +224,24 @@ naming scheme when pushing your branch(es):
...
-You should use your best judgement when facing such decisions. A good approach
+You should use your best judgment when facing such decisions. A good approach
 for this is to put yourself in the shoes of the person who will review your
 patch: will they understand your changes and reasoning just by reading your
 commit history? Will they find unrelated changes in a particular commit? They
 shouldn't!
 
 Writing meaningful commit messages that follow our commit message format will
-also help you respect this mantra (see the below section).
+also, help you respect this mantra (see the below section).
```
Comment (h3xar0n) [inline CONTRIBUTING.md]: This line is part of the sentence above, so this comma is incorrect. Please remove it.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -364,7 +361,7 @@ To inhibit the automatic addition of ToC, add the following to the front-matter
 `toc: false`
 
 This ToC generator depends on headings being correctly coded in the markdown
-portion of the doc site files. If a page has an incorrectly-formatted ToC, be
+the portion of the doc site files. If a page has an incorrectly-formatted ToC, be
```
Comment (h3xar0n) [inline CONTRIBUTING.md]: This line is part of the sentence above. The term is "the markdown portion of the doc site," so "the markdown the portion..." would not make sense. Please revert.

**Comment by levfishbluefish** [comment]
Thank you for contributing to Kong's documentation. However, this PR does not meet our criteria in order to be merged.

**Comment by h3xar0n** [comment]
@Soaibkhan38 , this PR has good improvements--please just implement the feedback.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Soaibkhan38) [reply CONTRIBUTING.md]: The content of that line was being shifted to the next line, so I shifted it back

File: CONTRIBUTING.md (hunk 3 above)
Comment (Soaibkhan38) [reply CONTRIBUTING.md]: Same as above, it was shifting to the new line.

File: CONTRIBUTING.md (hunk 4 above)
Comment (Soaibkhan38) [reply CONTRIBUTING.md]: Thanks for your comment, I resolved it back to original

**Comment by Soaibkhan38** [comment]
Thank You @h3xar0n for reviewing my changes. I have made several new commits to correct the file as per feedback. Please review my updates

**Comment by h3xar0n** [comment]
Thank you for your contribution, @Soaibkhan38
